client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java

```diff
@@ -492,14 +495,27 @@ private void reconnect(
             if (connectionFailureCause != null) {
                 log.debug("connection attempt failed", connectionFailureCause);
             }
-            final long reconnectInterval = Optional.ofNullable(clientOptions)
-                    .map(o -> o.getReconnectInterval())
-                    .orElse(Constants.DEFAULT_RECONNECT_INTERVAL_MILLIS);
-            log.trace("scheduling new connection attempt in {}ms ...", reconnectInterval);
-            // give Vert.x some time to clean up NetClient
-            vertx.setTimer(reconnectInterval, tid -> {
+            // apply exponential backoff with jitter
+            // determine the max delay for this reconnect attempt as 2^attempt * delayIncrement
+            final long currentMaxDelay = (long) Math.pow(2, reconnectAttempt - 1)
+                    * clientConfigProperties.getReconnectDelayIncrement();
+            final long reconnectInterval;
+            if (currentMaxDelay > clientConfigProperties.getReconnectMinDelay()) {
+                // let the actual reconnect delay be a random between the minDelay and the currentMaxDelay,
+                // capped by the overall maxDelay
```

Comment:
that is quite a spread, isn't it? it means that we wait for anything between 0 and 7000ms, right? The jitter is usually within 
95 and 105% (?) of the target value, just enough to make it not strictly deterministic ...

Reply:
It is, I was opting for trying to evenly spread out the calls to the stressed/restarting component (e.g. device registry) coming from different components (e.g. protocol adapters), taking into account that we rather have a shorter delay than a longer one. In that sense it's a kind of "full jitter" approach (see also [here](https://aws.amazon.com/de/blogs/architecture/exponential-backoff-and-jitter/)).